src/Native/Unix/System.Security.Cryptography.Native/pal_rsa.cpp

```diff
@@ -67,12 +94,31 @@ extern "C" int32_t CryptoNative_RsaGenerateKeyEx(RSA* rsa, int32_t bits, BIGNUM*
 extern "C" int32_t
 CryptoNative_RsaSign(int32_t type, const uint8_t* m, int32_t mlen, uint8_t* sigret, int32_t* siglen, RSA* rsa)
 {
-    if (!siglen)
+    if (siglen == nullptr)
     {
         assert(false);
         return 0;
     }
 
+    *siglen = 0;
```

Comment:
m is the message (digest) to sign, sigret is where to write it to, those are both ref, not out.